### PR TITLE
Enhance Profile Functionality

### PR DESCRIPTION
**Pull request by mohamdlog**
## Description

This PR introduces key enhancements to the profile section, focusing on managing user listings and implementing secure account deletion. The changes are aimed at improving user experience and ensuring data security.

### Backend

- `userController.js`
    - Created the file and implemented account deletion with password verification.
    - Ensured that all user-created items are deleted along with the account.
- `userModel.js`
    - Updated to prevent model re-initialization, resolving `OverwriteModelError`.
- `itemController.js`
    - Modified to support fetching user-specific listings.
- `userRoutes.js`
    - Created new routes for user-specific functionality.
- `server.js`
    - Updated to include and mount new user routes.

## Frontend

- Profile Sidebar
    - Updated to highlight the active page and prevent refreshes during navigation.
    - Implemented the `List an Item` feature using existing components.
- My Listings Page
    - Enhanced to display all items created by the authenticated user.
- Delete Account Page
    - Created a new page with a password confirmation input for secure account deletion.
    - Styled the component for a seamless user experience.

## Testing

- End-to-End Testing
    - Verified account creation, item creation, item deletion, and secure account deletion.
- User Experience
    - Confirmed smooth navigation in the profile sidebar.
    - Tested proper display and deletion of user listings.
 
## Summary
This PR enhances the profile functionalities, ensures secure account management, and provides a more user-friendly interface. All changes have been tested for reliability and security.

### CODE REVIEW

File: backend/controllers/userController.js
Comment (shagunsharmaslu) [inline backend/controllers/userController.js:1]: Nice and clean.

**Review by shagunsharmaslu — APPROVED** [review]
Look good to me. Approving.

File: backend/routes/itemRoutes.js [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 router.get('/items/:id', getItemById);
-router.post('/items', authMiddleware, createItem); // Protect this route with authMiddleware
-router.put('/items/:id', authMiddleware, updateItem); // Protect update if only logged-in users can update
-router.delete('/items/:id', authMiddleware, deleteItem); // Protect delete if only logged-in users can delete
+router.get('/my-items', authMiddleware, getMyItems);
+router.post('/items', authMiddleware, createItem);
+router.put('/items/:id', authMiddleware, updateItem);
+router.delete('/items/:id', authMiddleware, deleteItem);
 
 export default router;
```
Comment (nlp-saiteja) [inline backend/routes/itemRoutes.js:15]: This optimizes model usage by preventing redundant recompilation in userModel.js and improves route security in itemRoutes.js. A new getMyItems route allows authenticated users to fetch their own items, and authMiddleware is now consistently applied to protect create, update, and delete operations.

**Review by nlp-saiteja — APPROVED** [review]
looks good to me

**Pull request merged**